metric_learn/base_metric.py

```diff
@@ -369,6 +386,168 @@ def score(self, pairs, y):
 
+  def set_default_threshold(self, pairs, y):
+    """Sets the default threshold on the given dataset.
+
+    Returns a threshold that is the mean between the similar
+    metrics mean and the dissimilar metrics mean.
+
+    See more in the :ref:`User Guide <calibration>`.
+    """
+    similar_threshold = np.mean(self.score_pairs(
+        pairs[(y == 1).ravel()]))
+    dissimilar_threshold = np.mean(self.score_pairs(
+        pairs[(y == -1).ravel()]))
+    self.threshold_ = np.mean([similar_threshold, dissimilar_threshold])
+
+  def set_threshold(self, threshold):
+    """Sets the threshold of the metric learner to the given value `threshold
+
+    See more in the :ref:`User Guide <calibration>`.
+
+    Parameters
+    ----------
+    threshold : float
+      The threshold value we want to set. It's a distance metric with
+      respect to which the predicted distance metric for test pairs will be
+      compared to. If they are superior to the threshold they will be
+      classified as similar (+1), and dissimilar (-1) if not.
+
+    Returns
+    -------
+    self : `_PairsClassifier`
+      The pairs classifier with the new threshold set.
+    """
+    self.threshold_ = threshold
+    return self
+
+  def calibrate_threshold(self, pairs_valid, y_valid, strategy='accuracy',
+                          threshold=None, beta=None):
+    """Decision threshold calibration for binary classification
+
+    Method that calibrates the decision threshold (cutoff point) of the metric
+    learner. This threshold will then be used when calling the method
+    `predict`. The methods for picking cutoff points make use of traditional
+    binary classification evaluation statistics such as the true positive and
+    true negative rates and F-scores. The threshold will be found to maximize
+    the chosen score on the validation set `(pairs_valid, y_valid)`.
+
+    See more in the :ref:`User Guide <calibration>`.
+
+    Parameters
+    ----------
+    strategy : str, optional (default='roc')
+      The strategy to use for choosing the cutoff point
+
+      'accuracy'
+          selects a decision threshold that maximizes the accuracy
+      'f_beta'
+          selects a decision threshold that maximizes the f_beta score
+      'max_tpr'
+          selects the point that yields the highest true positive rate with
+          true negative rate at least equal to the value of the parameter
+          threshold
+      'max_tnr'
+          selects the point that yields the highest true negative rate with
+          true positive rate at least equal to the value of the parameter
+          threshold
+
+    beta : float in [0, 1], optional (default=None)
+      beta value to be used in case strategy == 'f_beta'
+
+    threshold : float in [0, 1] or None, (default=None)
+      In case strategy is 'max_tpr' or 'max_tnr' this parameter must be set
+      to specify the threshold for the true negative rate or true positive
```

Comment:
value instead of threshold

Reply:
Agreed, even `minimal value` would be good no ?

Reply:
I don't think so, as discussed in other comment?